amp.php

```diff
@@ -157,3 +160,20 @@ function _amp_bootstrap_customizer() {
 	}
 }
 add_action( 'plugins_loaded', '_amp_bootstrap_customizer', 9 );
+
+/**
+ * Redirects the old AMP URL to the new AMP URL.
+ * If post slug is updated the amp page with old post slug will be redirected to the updated url.
+ *
+ * @param  string $link New URL of the post.
+ *
+ * @return string $link URL to be redirected.
+ */
+function amp_redirect_to_new_url( $link ) {
+
+	if ( is_amp_endpoint() ) {
+		$link = trailingslashit( $link ) . AMP_QUERY_VAR;
```

Comment:
This would be nice if we get id also in filter. 

But about changing amp url can you please explain the issue because I can see that we are using the link as per modified permalink itself https://core.trac.wordpress.org/browser/tags/4.7/src/wp-includes/query.php#L896 . 

Also  we already added `amp_query_var` filter for `AMP_QUERY_VAR` so when the url is going to change this can also change. https://github.com/Automattic/amp-wp/blob/master/amp.php#L52

Please let me know if I am missing something.

Thank you:)